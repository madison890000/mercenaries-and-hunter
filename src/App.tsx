import React, { useRef } from 'react';
import styles from './App.module.scss';
import madison from './Madison';
import Header from './components/Header';
import Skill from './components/Skill';
import Period from './components/Period';
import Education from './components/Education';
import Divider from './components/Divider';
import { getPageContentWidth, getScreenDevice, ScreenDevice } from './utils/device';
import TimeLines from './components/TimeLines';
import Capability from './components/Capability';

const COLORS = [
    '#FF6666',
    '#FFFF66',
    '#66FFFF',
    '#6666FF',
    '#B266FF',
    '#FF66B2',
]

function App() {
    const { current: person } = useRef(madison);
    const periodColors: { [key: string]: string } = {};
    person?.periods?.forEach((p, index) => {
        periodColors[p.id] = COLORS[index];
    })
    return (
        <div className={styles.main}>
            <Header
                firstName={person.firstName}
                lastName={person.lastName}
                email={person.email}
                cellphone={person.cellphone}
            />
            <Divider title="Profile" />
            <section>
                <div className={styles.profile}>
                    <div className={styles.description}>
                        {person.description?.map(d => (
                            <div className={styles.descriptionItem}>{d}</div>
                        ))}
                    </div>
                    <div className={styles.capability}>
                        <Capability capabilities={person.capability} />
                    </div>

                </div>
            </section>
            <Divider title="Skill" />
            <section>
                <div className={styles.skills}>
                    {person.skills?.map(skill => (
                        <Skill {...skill} />
                    ))}
                </div>
            </section>
            <Divider
                title="Professional Experience"
                extra={
                    getScreenDevice() === ScreenDevice.PC ?
                        <TimeLines
                            width={getPageContentWidth() - 300}
                            periods={person.periods}
                            periodColors={periodColors}
                            barPosition="bottom"
                        />
                        : null
                }
            />
            <section>
                {
                    getScreenDevice() !== ScreenDevice.PC ? <TimeLines
                        width={getPageContentWidth()}
                        periods={person.periods}
                        periodColors={periodColors}
                        barPosition="top"
                    /> : null
                }
            </section>
            <section>
                {person.reversedPeriods?.map(period => (
                    <Period
                        {...period}
                        periodColor={periodColors[period?.id]}
                        companyName={period?.company?.name}
                        companyType={period?.company?.type}
                    />
                ))}
            </section>
            <Divider title="Education Experience" />
            <section>
                {person.educations?.map(education => (
                    <Education {...education} />
                ))}
            </section>
        </div>
    );
}

export default App;
