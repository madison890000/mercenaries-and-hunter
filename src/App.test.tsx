import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

xit('renders App', () => {
    const AppDom = renderer.create(<App />)
        .toJSON();
    expect(AppDom)
        .toMatchSnapshot();
});
