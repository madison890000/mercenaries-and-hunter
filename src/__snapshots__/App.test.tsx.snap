// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<div
  className="main"
>
  <header
    className="header"
  >
    <h1>
      Madison
       
      Ma
    </h1>
    <div
      className="basicInfo"
    >
      <div>
        <span
          className="email"
        >
          madison.sss.ma@gmail.com
        </span>
      </div>
      <div>
        <span
          className="cellphone"
        >
          +8618510163161
        </span>
      </div>
    </div>
  </header>
  <div
    className="divider"
  >
    <h3>
      Description
    </h3>
    <div
      className="section"
    />
  </div>
  <section>
    <div
      className="description"
    >
      <div
        className="descriptionItem"
      >
        I have 7 years of working experience in IT with solid technical background, proficient in modern JavaScript/TypeScript programming frameworks, tools, and libraries. 
      </div>
      <div
        className="descriptionItem"
      >
        I Have rich experience in Front-End and Back-End development including Admin Portal and Customer Portal in Online Sale Web
Application, Internet finance Web System, education application, Data Reporting Web Page, etc.
      </div>
      <div
        className="descriptionItem"
      >
        I am both a good team leader and team member. I am a person who is passionate about life, friendly to people, willing to share, and like to work with colleagues to make the world to be a better place.
      </div>
    </div>
  </section>
  <div
    className="divider"
  >
    <h3>
      Skill
    </h3>
    <div
      className="section"
    />
  </div>
  <section>
    <div
      className="skills"
    >
      <div
        className="skill"
      >
        <div>
          <div
            className="important"
          >
            JavaScript
          </div>
          <div
            className="subTitle"
          >
            7
             years
          </div>
        </div>
        <div
          className="level proficient"
        >
          <span />
          <span />
          <span />
          <div
            className="levelName"
          >
            Proficient
          </div>
        </div>
      </div>
      <div
        className="skill"
      >
        <div>
          <div
            className="important"
          >
            React
          </div>
          <div
            className="subTitle"
          >
            5
             years
          </div>
        </div>
        <div
          className="level proficient"
        >
          <span />
          <span />
          <span />
          <div
            className="levelName"
          >
            Proficient
          </div>
        </div>
      </div>
      <div
        className="skill"
      >
        <div>
          <div
            className="important"
          >
            TypeScript
          </div>
          <div
            className="subTitle"
          >
            3
             years
          </div>
        </div>
        <div
          className="level proficient"
        >
          <span />
          <span />
          <span />
          <div
            className="levelName"
          >
            Proficient
          </div>
        </div>
      </div>
      <div
        className="skill"
      >
        <div>
          <div
            className="important"
          >
            React hooks
          </div>
          <div
            className="subTitle"
          >
            3
             years
          </div>
        </div>
        <div
          className="level proficient"
        >
          <span />
          <span />
          <span />
          <div
            className="levelName"
          >
            Proficient
          </div>
        </div>
      </div>
      <div
        className="skill"
      >
        <div>
          <div
            className="important"
          >
            React-Redux
          </div>
          <div
            className="subTitle"
          >
            3
             years
          </div>
        </div>
        <div
          className="level proficient"
        >
          <span />
          <span />
          <span />
          <div
            className="levelName"
          >
            Proficient
          </div>
        </div>
      </div>
      <div
        className="skill"
      >
        <div>
          <div
            className="important"
          >
            HTML
          </div>
          <div
            className="subTitle"
          >
            6
             years
          </div>
        </div>
        <div
          className="level proficient"
        >
          <span />
          <span />
          <span />
          <div
            className="levelName"
          >
            Proficient
          </div>
        </div>
      </div>
      <div
        className="skill"
      >
        <div>
          <div
            className="important"
          >
            CSS
          </div>
          <div
            className="subTitle"
          >
            6
             years
          </div>
        </div>
        <div
          className="level proficient"
        >
          <span />
          <span />
          <span />
          <div
            className="levelName"
          >
            Proficient
          </div>
        </div>
      </div>
      <div
        className="skill"
      >
        <div>
          <div
            className="normal"
          >
            MobX
          </div>
          <div
            className="subTitle"
          >
            1
             years
          </div>
        </div>
        <div
          className="level proficient"
        >
          <span />
          <span />
          <span />
          <div
            className="levelName"
          >
            Proficient
          </div>
        </div>
      </div>
      <div
        className="skill"
      >
        <div>
          <div
            className="normal"
          >
            Electron
          </div>
          <div
            className="subTitle"
          >
            1
             years
          </div>
        </div>
        <div
          className="level familiar"
        >
          <span />
          <span />
          <span />
          <div
            className="levelName"
          >
            Familiar
          </div>
        </div>
      </div>
      <div
        className="skill"
      >
        <div>
          <div
            className="normal"
          >
            Node
          </div>
          <div
            className="subTitle"
          >
            2
             years
          </div>
        </div>
        <div
          className="level familiar"
        >
          <span />
          <span />
          <span />
          <div
            className="levelName"
          >
            Familiar
          </div>
        </div>
      </div>
      <div
        className="skill"
      >
        <div>
          <div
            className="normal"
          >
            Vue
          </div>
          <div
            className="subTitle"
          >
            1.5
             years
          </div>
        </div>
        <div
          className="level proficient"
        >
          <span />
          <span />
          <span />
          <div
            className="levelName"
          >
            Proficient
          </div>
        </div>
      </div>
      <div
        className="skill"
      >
        <div>
          <div
            className="normal"
          >
            Java
          </div>
          <div
            className="subTitle"
          >
            3
             years
          </div>
        </div>
        <div
          className="level familiar"
        >
          <span />
          <span />
          <span />
          <div
            className="levelName"
          >
            Familiar
          </div>
        </div>
      </div>
      <div
        className="skill"
      >
        <div>
          <div
            className="less"
          >
            Docker
          </div>
          <div
            className="subTitle"
          >
            0.5
             years
          </div>
        </div>
        <div
          className="level understand"
        >
          <span />
          <span />
          <span />
          <div
            className="levelName"
          >
            Understand
          </div>
        </div>
      </div>
      <div
        className="skill"
      >
        <div>
          <div
            className="less"
          >
            C#
          </div>
          <div
            className="subTitle"
          >
            0.5
             years
          </div>
        </div>
        <div
          className="level understand"
        >
          <span />
          <span />
          <span />
          <div
            className="levelName"
          >
            Understand
          </div>
        </div>
      </div>
      <div
        className="skill"
      >
        <div>
          <div
            className="normal"
          >
            Database
          </div>
          <div
            className="subTitle"
          >
            2
             years
          </div>
        </div>
        <div
          className="level familiar"
        >
          <span />
          <span />
          <span />
          <div
            className="levelName"
          >
            Familiar
          </div>
        </div>
      </div>
      <div
        className="skill"
      >
        <div>
          <div
            className="important"
          >
            Git
          </div>
          <div
            className="subTitle"
          >
            6
             years
          </div>
        </div>
        <div
          className="level proficient"
        >
          <span />
          <span />
          <span />
          <div
            className="levelName"
          >
            Proficient
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    className="divider"
  >
    <h3>
      Professional Experience
    </h3>
    <div
      className="section"
    />
  </div>
  <section>
    <div
      className="timelineContainer"
    >
      <div>
        <svg
          height="20"
          style={
            Object {
              "backgroundColor": "E0E0E0",
            }
          }
          width={0}
        >
          <g>
            <rect
              fill="#FF6666"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FF6666"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FF6666"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FF6666"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FF6666"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FF6666"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FF6666"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FF6666"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FF6666"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FF6666"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FF6666"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FF6666"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FF6666"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#FFFF66"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#66FFFF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#66FFFF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#66FFFF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#66FFFF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#66FFFF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#66FFFF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#66FFFF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#66FFFF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#66FFFF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#66FFFF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#66FFFF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#66FFFF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#66FFFF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#66FFFF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#66FFFF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#6666FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#B266FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#B266FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#B266FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
            <rect
              fill="#B266FF"
              height="10"
              width={-1}
              x={0}
              y="5"
            />
          </g>
        </svg>
      </div>
      <div
        style={
          Object {
            "height": 48,
          }
        }
      >
        <span
          className="jobName"
          style={
            Object {
              "position": "absolute",
              "transform": "translateX(0px)",
            }
          }
        >
          Junior
           
          Software Engineer
        </span>
        <span
          className="jobName"
          style={
            Object {
              "position": "absolute",
              "transform": "translateX(0px)",
            }
          }
        >
           
          Software Engineer
        </span>
        <span
          className="jobName"
          style={
            Object {
              "position": "absolute",
              "transform": "translateX(0px)",
            }
          }
        >
           
          Front-End Engineer
        </span>
        <span
          className="jobName"
          style={
            Object {
              "position": "absolute",
              "transform": "translateX(0px)",
            }
          }
        >
          Senior
           
          Front-End Engineer
        </span>
        <span
          className="jobName"
          style={
            Object {
              "position": "absolute",
              "transform": "translateX(0px)",
            }
          }
        >
          Senior
           
          Software Engineer
        </span>
      </div>
    </div>
  </section>
  <section>
    <div
      className="period"
    >
      <div>
        <div
          className="header"
        >
          <span
            className="job"
          >
            <span>
              Senior
            </span>
            <span>
              Software Engineer
            </span>
          </span>
          <span
            className="companyName"
          >
            <span>
              (
              Tesla
              )
            </span>
          </span>
          <span>
            <span
              className="keyword"
            >
              React
            </span>
          </span>
        </div>
      </div>
      <div
        className="times"
      >
        <span>
          May, 2022
        </span>
        <span>
          -
        </span>
        <span>
          Sep, 2022
        </span>
      </div>
    </div>
    <div>
      <h5>
        Job summaries:
      </h5>
      <ul>
        <li>
          make the architecture design in Front-End project
        </li>
        <li>
          improve the code quality and performance in FE
        </li>
        <li>
          improve the FE team efficiency
        </li>
        <li>
          communicate with product manager and UI designer to understand project requirements every 2 week (one sprint),
        </li>
        <li>
          provide training and guidance on technical
        </li>
      </ul>
    </div>
    <div>
      <h5>
        Achievements:
      </h5>
      <ul
        className="achievements"
      >
        <li>
          <span>
            <span
              className="category"
            >
              Engineering
            </span>
          </span>
          completed the design and development of common utils and Hooks in React project
        </li>
        <li>
          <span>
            <span
              className="category"
            >
              Performance
            </span>
          </span>
          optimized the front-end code
        </li>
        <li>
          <span>
            <span
              className="category"
            >
              Quality
            </span>
          </span>
          improve the code quality
        </li>
      </ul>
    </div>
    <div
      className="period"
    >
      <div>
        <div
          className="header"
        >
          <span
            className="job"
          >
            <span>
              Senior
            </span>
            <span>
              Front-End Engineer
            </span>
          </span>
          <span
            className="companyName"
          >
            <span>
              (
              PCCW GLOBAL
              )
            </span>
          </span>
          <span>
            <span
              className="keyword"
            >
              React
            </span>
            <span
              className="keyword"
            >
              NextJs
            </span>
            <span
              className="keyword"
            >
              umi
            </span>
            <span
              className="keyword"
            >
              TDD
            </span>
          </span>
        </div>
      </div>
      <div
        className="times"
      >
        <span>
          Jun, 2019
        </span>
        <span>
          -
        </span>
        <span>
          May, 2022
        </span>
      </div>
    </div>
    <div>
      <h5>
        Job summaries:
      </h5>
      <ul>
        <li>
          make the architecture design in Front-End project
        </li>
        <li>
          communicate with product manager and UI designer to understand project requirements every week (one sprint),
        </li>
        <li>
          conduct functional research and validation based on business requirements, evaluate requirements implementation feasibility and propose solutions.
        </li>
        <li>
          responsible for task assignment and project requirement explanation to 5 external team members
        </li>
        <li>
          provide training and guidance on technical to junior engineer
        </li>
      </ul>
    </div>
    <div>
      <h5>
        Achievements:
      </h5>
      <ul
        className="achievements"
      >
        <li>
          <span>
            <span
              className="category"
            >
              UI
            </span>
          </span>
          design and developed our company's front-end UI component library
        </li>
        <li>
          <span>
            <span
              className="category"
            >
              CI/CD
            </span>
            <span
              className="category"
            >
              Engineering
            </span>
          </span>
          setup the Front-End CI/CD with system Engineers, including the Docker building and the jest process
        </li>
        <li>
          <span>
            <span
              className="category"
            >
              Engineering
            </span>
          </span>
          setup the utils library in FE
        </li>
        <li>
          <span>
            <span
              className="category"
            >
              Performance
            </span>
            <span
              className="category"
            >
              Engineering
            </span>
          </span>
          using Micro Front-End Service to split code and improve the performance in FE
        </li>
        <li>
          <span>
            <span
              className="category"
            >
              Architecture
            </span>
          </span>
          define the Best practices in FE project with React and Typescript.
        </li>
      </ul>
    </div>
    <div
      className="period"
    >
      <div>
        <div
          className="header"
        >
          <span
            className="job"
          >
            <span>
              Front-End Engineer
            </span>
          </span>
          <span
            className="companyName"
          >
            <span>
              (
              Citic Baixin Bank Co., Ltd
              )
            </span>
          </span>
          <span>
            <span
              className="keyword"
            >
              Vue
            </span>
            <span
              className="keyword"
            >
              React
            </span>
          </span>
        </div>
      </div>
      <div
        className="times"
      >
        <span>
          Apr, 2018
        </span>
        <span>
          -
        </span>
        <span>
          Jun, 2019
        </span>
      </div>
    </div>
    <div>
      <h5>
        Job summaries:
      </h5>
      <ul>
        <li>
          communicate with product manager and UI designer partner through meeting to understand project requirements
        </li>
        <li>
          responsible for task assignment and project requirement explanation to 5 team members
        </li>
        <li>
          provide training and guidance on technical to junior engineer
        </li>
      </ul>
    </div>
    <div>
      <h5>
        Achievements:
      </h5>
      <ul
        className="achievements"
      >
        <li>
          <span>
            <span
              className="category"
            >
              UI
            </span>
          </span>
          completed the design and development of company's front-end UI component library
        </li>
      </ul>
    </div>
    <div
      className="period"
    >
      <div>
        <div
          className="header"
        >
          <span
            className="job"
          >
            <span>
              Software Engineer
            </span>
          </span>
          <span
            className="companyName"
          >
            <span>
              (
              Yinyuan Asia-pacific (Beijing) Technology Co., Ltd 
              )
            </span>
          </span>
          <span>
            <span
              className="keyword"
            >
              Java
            </span>
            <span
              className="keyword"
            >
              jQuery
            </span>
            <span
              className="keyword"
            >
              React
            </span>
          </span>
        </div>
      </div>
      <div
        className="times"
      >
        <span>
          Jul, 2016
        </span>
        <span>
          -
        </span>
        <span>
          Mar, 2018
        </span>
      </div>
    </div>
    <div>
      <h5>
        Job summaries:
      </h5>
      <ul>
        <li>
          participate in the design and development of all projects 
        </li>
        <li>
          responsible for the release quality of all system. 
        </li>
        <li>
          responsible for team's recruitment, technical training(ES6), etc. 
        </li>
        <li>
          take charge of annual feature planning and appraise performance of each team member.
        </li>
      </ul>
    </div>
    <div>
      <h5>
        Achievements:
      </h5>
      <ul
        className="achievements"
      >
        <li>
          <span>
            <span
              className="category"
            >
              UI
            </span>
          </span>
          completed the design and development of the company's front-end UI component library
        </li>
        <li>
          <span>
            <span
              className="category"
            >
              UI
            </span>
          </span>
          'implemented the rendering of list components and displaying data  in both horizontal and vertical
        </li>
      </ul>
    </div>
    <div
      className="period"
    >
      <div>
        <div
          className="header"
        >
          <span
            className="job"
          >
            <span>
              Junior
            </span>
            <span>
              Software Engineer
            </span>
          </span>
          <span
            className="companyName"
          >
            <span>
              (
              Taiji Computer Corporation Limited
              )
            </span>
          </span>
          <span>
            <span
              className="keyword"
            >
              Java
            </span>
            <span
              className="keyword"
            >
              jQuery
            </span>
          </span>
        </div>
      </div>
      <div
        className="times"
      >
        <span>
          Jul, 2015
        </span>
        <span>
          -
        </span>
        <span>
          Jul, 2016
        </span>
      </div>
    </div>
    <div>
      <h5>
        Job summaries:
      </h5>
      <ul>
        <li>
          use Java to develop system on schedule
        </li>
        <li>
          participate in the design and development of one economic-data-report project.
        </li>
      </ul>
    </div>
    <div>
      <h5>
        Achievements:
      </h5>
      <ul
        className="achievements"
      >
        <li>
          <span>
            <span
              className="category"
            >
              UI
            </span>
          </span>
          abstract 10+ components to unify UI style of most company projects, including table, form input, file upload, etc
        </li>
      </ul>
    </div>
  </section>
  <div
    className="divider"
  >
    <h3>
      Education Experience
    </h3>
    <div
      className="section"
    />
  </div>
  <section>
    <div
      className="education"
    >
      <div>
        <span>
          China Agricultural University
        </span>
      </div>
      <div>
        <span>
          Resources and Environment
        </span>
        <span
          className="major"
        >
          (
          Bachelor
          )
        </span>
      </div>
      <div
        className="times"
      >
        <span>
          Sep, 2008
        </span>
        <span>
          -
        </span>
        <span>
          Jul, 2012
        </span>
      </div>
    </div>
    <div
      className="education"
    >
      <div>
        <span>
          China Agricultural University
        </span>
      </div>
      <div>
        <span>
          Geographical Information System
        </span>
        <span
          className="major"
        >
          (
          Master
          )
        </span>
      </div>
      <div
        className="times"
      >
        <span>
          Sep, 2012
        </span>
        <span>
          -
        </span>
        <span>
          Jul, 2015
        </span>
      </div>
    </div>
  </section>
</div>
`;
